### PR TITLE
Use `knative.dev/pkg/changeset`

### PR DESCRIPTION
**Pull request by mattmoor**
# Changes

This replaces the need for `./pkg/version` and a flag with the automatic inference of the version of the deployed code from the git metadata symlinked into `kodata/`

This package is already used to automatically decorate the zap loggers, and a few other places (iirc).

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @vdemeester @afrittoli @imjasonh @dlorenc

**Comment by imjasonh** [comment]
Is the change to controller.yaml going to run afoul of the modification done in

https://github.com/tektoncd/pipeline/blob/a728ce3c3d717fe52ff0b1eb71d357460498115c/tekton/publish.yaml#L131-L132

**Comment by mattmoor** [comment]
Thanks for making my eyes bleed @imjasonh 😉 

There's probably an expression in there we no longer need, but I hate touching bash that presubmits don't check.  Lemme look closer.

**Comment by mattmoor** [comment]
@imjasonh @vdemeester this should be RFAL.

**Comment by mattmoor** [comment]
post-weekend bump 😁

**Comment by vdemeester** [comment]
@mattmoor does it support only a commit hash, or can it show a tag ?

**Comment by mattmoor** [comment]
> does it support only a commit hash, or can it show a tag

As it is today, I believe it resolves the tag to a commit, but all the info is in kodata, so we could adapt the changeset package to do more.

**Review by vdemeester — APPROVED** [review]

**Comment by vdemeester** [comment]
So @mattmoor one *downside* of this approach is, it fails if the images are not build with `ko` 😛 (which is the case for OpenShift Pipelines for example, we rebuild them differently). https://github.com/knative/pkg/blob/main/changeset/commit.go#L60 is *less* forgiving than what I initially though 😛

File: pkg/pod/pod.go [hunk 1]
```diff
@@ -290,7 +290,11 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 	}
 
 	podAnnotations := taskRun.Annotations
-	podAnnotations[ReleaseAnnotation] = version.PipelineVersion
+	version, err := changeset.Get()
+	if err != nil {
+		return nil, err
+	}
+	podAnnotations[ReleaseAnnotation] = version
```
Comment (vdemeester) [inline pkg/pod/pod.go:297]: Maybe we should log something and print `unknown` here instead of erroring out, wdyt @mattmoor ?

**Pull request merged**